browser/ui/webui/settings/brave_default_extensions_handler.cc

```diff
@@ -44,6 +55,49 @@ void BraveDefaultExtensionsHandler::RegisterMessages() {
       base::BindRepeating(
         &BraveDefaultExtensionsHandler::SetIPFSCompanionEnabled,
         base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "setMediaRouterEnabled",
+      base::BindRepeating(
+        &BraveDefaultExtensionsHandler::SetMediaRouterEnabled,
+        base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      flags_ui::kRestartBrowser,
+      base::BindRepeating(&BraveDefaultExtensionsHandler::HandleRestartBrowser,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "getRestartNeeded",
+      base::BindRepeating(
+        &BraveDefaultExtensionsHandler::GetRestartNeeded,
+        base::Unretained(this)));
+}
+
+void BraveDefaultExtensionsHandler::GetRestartNeeded(
+    const base::ListValue* args) {
+  bool media_router_current_pref = profile_->GetPrefs()->GetBoolean(
+      prefs::kEnableMediaRouter);
+  bool media_router_new_pref = profile_->GetPrefs()->GetBoolean(
+      kBraveEnabledMediaRouter);
+  bool media_router_enabled = extensions::ExtensionPrefs::Get(profile_)->
+      IsExtensionDisabled(extension_misc::kMediaRouterStableExtensionId);
+
+  // This only happens on upgrade if the user has enabled media router using
+  // brave://flags before. We explicitly require the user to enable the pref
+  // to avoid any surprises about new connections on upgrade.
+  if (media_router_enabled != media_router_current_pref) {
+    restart_needed_ = (media_router_current_pref == media_router_new_pref);
+  } else {
+    restart_needed_ = (media_router_current_pref != media_router_new_pref);
+  }
+
+
+  AllowJavascript();
+  ResolveJavascriptCallback(args->GetList()[0].Clone(),
```

Comment:
why are you cloning this? Also `args->GetList()[0]` is not the correct or safe way to get this value. 
```
std::string callback_id;
args->GetString(0, &callback_id);
```